subsys/bluetooth/mesh/pb_adv.c

```diff
@@ -317,6 +359,142 @@ static void protocol_timeout(struct k_work *work)
 	link.rx.seg = 0U;
 	prov_link_close(PROV_BEARER_LINK_STATUS_TIMEOUT);
 }
+
+static struct delayable_adv_ctx *delayable_adv_ctx_get_unused(void)
+{
+	for (int i = 0; i < CONFIG_BT_MESH_PB_ADV_DELAYABLE_ADV_COUNT; i++) {
+		if (!atomic_test_bit(delayable_advs_ctx[i].flags, DELAYABLE_ADV_CTX_IN_USE)) {
+			return &delayable_advs_ctx[i];
+		}
+	}
+
+	return NULL;
+}
+
+static struct delayable_adv_ctx *delayable_adv_ctx_assign(struct bt_mesh_adv *adv,
+							  const struct bt_mesh_send_cb *cb,
+							  void *cb_data, bool retransmit)
+{
+	struct delayable_adv_ctx *ctx;
+
+	ctx = delayable_adv_ctx_get_unused();
+	if (!ctx) {
+		LOG_WRN("Purge pending delayed advertisement.");
+		if (!delayable_adv_ctx_push()) {
```

Comment:
This brakes the shall statement of the spec regarding random delay. Can we allocate enough buffers to avoid this situation?

Reply:
This was added as just in case we have 2 acks and a full tx buffer in the delay. Should we increase the size of buffer? This was added as a just in case mechanism, if we do not want it we can remove it, and then just return NULL. That will just make this resend the messages, that it could not send.

Reply:
I do not think it breaks shall statement. It has already been delayed for some time. Unexpectedly completed random delay is still random. Unexpected completion is random act regarding protocol timings.